models/user.modal.go

```diff
@@ -0,0 +1,83 @@
+package models
+
+import (
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+)
+
+type User struct {
+	ID        *uuid.UUID `gorm:"type:uuid;default:uuid_generate_v4();primary_key"`
+	FirstName string     `gorm:"type:varchar(100);not null"`
+	LastName  string     `gorm:"type:varchar(100);not null"`
+	Username  string     `gorm:"type:varchar(100);uniqueIndex;not null"`
+	Role      *string    `gorm:"type:varchar(50);default:'user';not null"`
```

Comment:
do we need Role here..?

Reply:
as we have discussed over call we can add role in future